packages/medusa/src/services/region.ts

```diff
@@ -119,7 +145,8 @@ class RegionService extends BaseService {
           )
         }
 
-        regionObject.currency = currency
+        ;(regionObject as CreateRegionInput & { currency: Currency }).currency =
```

Comment:
todo: We should create a new object to assign the data instead of mutating the reference to the given object. It will also fix the type issue here

Reply:
comment: same for the other assignation of this method